lib/middlewares/pre_compressed.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+const { getType } = require('mime');
+
+module.exports = function(app) {
+  const { config, route } = this;
+  const { root } = config;
+
+  if (!config.server.preCompressed) return;
+
+  app.use(root, (req, res, next) => {
+    const { headers, method } = req;
+    const routeList = route.list();
+    const url = decodeURIComponent(req.url);
+    const acceptEncoding = headers['accept-encoding'] || '';
+    const reqUrl = url.endsWith('/') ? url.concat('index.html') : url;
```

Comment:
hexo-server doesn't support `example.com/foo -> example.com/foo.html` route, so it's not added here.